Drop WTF_ALLOW_UNSAFE_BUFFER_USAGE use in TextCodecCJK.cpp

cdumez opened the pull request:
#### a8a5ce06c98677e1746c7d67bc94381400ce1aa9
<pre>
Drop WTF_ALLOW_UNSAFE_BUFFER_USAGE use in TextCodecCJK.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=294418">https://bugs.webkit.org/show_bug.cgi?id=294418</a>

Reviewed by Darin Adler.

* Source/WebCore/PAL/pal/text/TextCodecCJK.cpp:
(PAL::gb18030RangesCodePoint):
(PAL::gb18030RangesPointer):

Canonical link: <a href="https://commits.webkit.org/296210@main">https://commits.webkit.org/296210@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6c3aee488137673b13b6253c72ef37f55b9785b1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/107681 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27355 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/17770 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/112891 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58217 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/109636 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28047 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/35925 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81760 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110615 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22229 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97047 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62139 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21663 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15188 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/57653 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91601 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15220 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116012 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/34757 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/25640 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90800 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/35134 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93297 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90551 "Passed tests") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23084 "Found 1 new test failure: http/tests/workers/service/registerServiceWorkerClient-after-network-process-crash.html (failure)") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35467 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13234 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/30512 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17421 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34662 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/40217 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34407 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/37768 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36070 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/PAL/pal/text/TextCodecCJK.cpp:
```diff
@@ -855,19 +855,19 @@ static const std::array<std::pair<uint32_t, char32_t>, 207>& gb18030Ranges()
+    ASSERT(upperBound != ranges.begin());
+    auto& range = ranges[upperBound - ranges.begin() - 1];
+    uint32_t offset = range.first;
+    char32_t codePointOffset = range.second;
```
Does something like this work?
```suggestion
    auto [offset, codePointOffset] = ranges[upperBound - ranges.begin() - 1];
```

cdumez replied:
Yes it does. I'll update the PR.

darinadler commented on Source/WebCore/PAL/pal/text/TextCodecCJK.cpp:
```diff
@@ -876,15 +876,15 @@ static uint32_t gb18030RangesPointer(char32_t codePoint)
+    ASSERT(upperBound != ranges.begin());
+    auto& range = ranges[upperBound - ranges.begin() - 1];
+    uint32_t pointerOffset = range.first;
+    char32_t offset = range.second;
```
Does something like this work?
```suggestion
    auto [pointerOffset, offset] = ranges[upperBound - ranges.begin() - 1];
```

cdumez committed:
Drop WTF_ALLOW_UNSAFE_BUFFER_USAGE use in TextCodecCJK.cpp
https://bugs.webkit.org/show_bug.cgi?id=294418

Reviewed by Darin Adler.

* Source/WebCore/PAL/pal/text/TextCodecCJK.cpp:
(PAL::gb18030RangesCodePoint):
(PAL::gb18030RangesPointer):

Canonical link: https://commits.webkit.org/296210@main
